File: cpp/593_ValidSquare.cpp
class Solution {
public:
    bool validSquare(vector<int>& p1, vector<int>& p2, vector<int>& p3, vector<int>& p4) {
        vector<vector<int>> points = {p1, p2, p3, p4};
        unordered_set<int> distance_set;
        for (int i = 0; i < points.size(); ++i) {
            for (int j = i + 1; j < points.size(); ++j) {
                if (points[i] == points[j]) 
                    return false;
                
                distance_set.insert(get_distance(points[i], points[j]));
            }
        }
        return distance_set.size() == 2;
    }
    
    int get_distance(const vector<int>& a, const vector<int>& b) {
        return (b[1] - a[1]) * (b[1] - a[1]) + (b[0] - a[0]) * (b[0] - a[0]);
    }
};

File: python/593_ValidSquare.py
class Solution:
    def validSquare(self, p1: List[int], p2: List[int], p3: List[int], p4: List[int]) -> bool:
        
        def get_dist(a, b):
            return (b[1] - a[1]) * (b[1] - a[1]) + (b[0] - a[0]) * (b[0] - a[0])
        
        points = [p1, p2, p3, p4]
        distance_set = {get_dist(points[i], points[j]) for i in range(len(points)) for j in range(i+1, len(points))}
        
        return len(distance_set) == 2 and 0 not in distance_set
